docs/src/views/Architecture.tsx

```diff
@@ -0,0 +1,224 @@
+        together and handles component properties, state and events, as ell as provides ability to
+        inject and handle themes, accessibility behaviors and accessibility keyboard navigation
+        handling.
+      </li>
+      <li>
+        <strong>Accessibility behaviors</strong> - the logic which generates appropriate
+        accessibility roles and <code>aria-*</code> attributes for the given component based on the
+        component state.
+      </li>
+      <li>
+        <strong>Accessibility keyboard handlers</strong> - provides the logic to handle components
+        navigation and focus.
+      </li>
+      <li>
+        <strong>Themes</strong> - the logic to inject and handle component visual representation
+        (theme).
+      </li>
+      <li>
+        <strong>Renderer</strong> - to render the HTML Output. Current implementation uses React,
+        but it can be extended to other frameworks.
+      </li>
+    </ul>
+    <p>
+      <img src="assets/stardust_arch_component_logic.png" />
+    </p>
+
+    <Header as="h3">Accessibility</Header>
+    <p>
+      One of the main purposes of the library is to make component accessible and provide ease
+      accessibility integration for the entire projects which will be consumer of the library.
+    </p>
+    <p>
+      The library encapsulates accessibility into <strong>Behaviors</strong>. Behavior is the
+      object, which has three major responsibilities:
+    </p>
+    <ul>
+      <li>
+        Define <code>aria-*</code> attributes for the standard components (e.g. Image, DropDown,
+        Dialog, etc), based on their state (e.g. active, disabled, expanded, etc).
+      </li>
+      <li>
+        Define the correct element <code>role</code>, if the HTML element tag differ from standard
+        (e.g. use <code>&lt;span&rt;</code> instead of <code>&lt;button&rt;</code>) to represent the
+        button element.
+      </li>
+      <li>
+        Define the keyboard navigation and focus handling for the composed components (e.g. List,
+        Grid, Radio Group, etc).
+      </li>
+    </ul>
+    <p>
+      Library provides the set of accessibility behaviors and keyboard handlers for the components
+      and allows to create and inject your own, if it is necessary for some custom components. In
+      most cases it should be enough to use the behaviors provided in the library, but, of course,
+      you can provide your own or extend existing behaviors.
+    </p>
+    <p>
+      In addition to behaviors and keyboard handlers, the toolkit also provides FocusZone component,
+      which can be used to add focus handling for the wrapped component. There is also FocusTrap
+      component, which allows to trap the focus inside some area, which is useful for such a
+      component as Dialog.
+    </p>
+
+    <Header as="h3">Theming</Header>
+    <p>
+      You can inject different themes into Stardust components (e.g. Light, Dark or High Contrast).
```

Comment:
Components cannot receive themes, only the Provider can receive a theme.  Components can only receive variables and props.